### PR TITLE
Add Twilio content template type

### PR DESCRIPTION
**Pull request by norkans7**

### CODE REVIEW

File: temba/templates/types/twilio/type.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+from temba.channels.models import Channel
+from temba.request_logs.models import HTTPLog
+
+from ...models import TemplateTranslation, TemplateType
+
+
+class TwilioType(TemplateType):
+    slug = "whatsapp"
+    variable_regex = re.compile(r"{{(\d+)}}")
```
Comment (rowanseymour) [inline temba/templates/types/twilio/type.py:16]: Twilio allows alphanumeric variables https://www.twilio.com/docs/content/using-variables-with-content-api

File: temba/templates/types/twilio/type.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+                            {
+                                "type": "button/quick_reply",
+                                "name": button_name,
+                                "content": button_text,
+                                "variables": button_vars,
+                                "params": [{"type": "text"} for v in button_vars],  # deprecated
+                            }
+                        )
+                    elif content_type == "twilio/call-to-action":
+                        button_type = action["type"].upper()
```
Comment (rowanseymour) [inline temba/templates/types/twilio/type.py:122]: unless the Twilio docs say that type is sometimes one case and sometimes another.. don't add ambiguity

**Review by rowanseymour — CHANGES_REQUESTED** [review]

File: temba/templates/types/twilio/type.py [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+from temba.channels.models import Channel
+from temba.request_logs.models import HTTPLog
+
+from ...models import TemplateTranslation, TemplateType
+
+
+class TwilioType(TemplateType):
+    slug = "whatsapp"
+    variable_regex = re.compile(r"{{[ ]*([A-Za-z0-9]+)[ ]*}}")
```
Comment (rowanseymour) [inline temba/templates/types/twilio/type.py:16]: it specifically says on https://www.twilio.com/docs/content/using-variables-with-content-api that they can't have spaces

(also `[ ]*` is not how you match whitespace in a regex)

Comment (norkans7) [reply temba/templates/types/twilio/type.py]: updated

File: temba/templates/types/twilio/tests.py [hunk 4]
```diff
@@ -0,0 +1,427 @@
...
+            [
+                {
+                    "type": "body",
+                    "name": "body",
+                    "content": "Welcome {{1}}, we have new features such as {{3}}, Subscribe for more info",
+                    "variables": {
+                        "1": 0,
+                        "3": 1,
+                    },
+                    "params": [{"type": "text"}, {"type": "text"}],
```
Comment (rowanseymour) [inline temba/templates/types/twilio/tests.py:340]: `"params"` ?

**Review by rowanseymour — CHANGES_REQUESTED** [review]

File: temba/templates/types/twilio/tests.py (hunk 4 above)
Comment (norkans7) [reply temba/templates/types/twilio/tests.py]: I just removed that, that was deprecated and I thought we needed that until the editor was updated, the editor has not been already updated now.

**Review by rowanseymour — APPROVED** [review]

**Pull request merged**